x-pack/legacy/plugins/apm/public/components/app/ErrorGroupOverview/index.tsx

```diff
@@ -22,73 +23,124 @@ import { ErrorDistribution } from '../ErrorGroupDetails/Distribution';
 import { ErrorGroupList } from './List';
 import { useUrlParams } from '../../../hooks/useUrlParams';
 import { useTrackPageview } from '../../../../../infra/public';
+import { PROJECTION } from '../../../projections/typings';
+import { LocalUIFilters } from '../../shared/LocalUIFilters';
+import { useTransactionTypes } from '../../../hooks/useTransactionTypes';
 
 const ErrorGroupOverview: React.SFC = () => {
+  const { urlParams, uiFilters } = useUrlParams();
+
   const {
-    urlParams: { serviceName, start, end, sortField, sortDirection },
-    uiFilters
-  } = useUrlParams();
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    transactionType
+  } = urlParams;
 
   const { data: errorDistributionData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorDistribution({
         serviceName,
+        transactionType,
         start,
         end,
         uiFilters
       });
     }
-  }, [serviceName, start, end, uiFilters]);
+  }, [serviceName, start, end, uiFilters, transactionType]);
 
   const { data: errorGroupListData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorGroupList({
         serviceName,
+        transactionType,
         start,
         end,
         sortField,
         sortDirection,
         uiFilters
       });
     }
-  }, [serviceName, start, end, sortField, sortDirection, uiFilters]);
+  }, [
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    uiFilters,
+    transactionType
+  ]);
+
+  const transactionTypes = useTransactionTypes(urlParams);
 
-  useTrackPageview({ app: 'apm', path: 'error_group_overview' });
+  useTrackPageview({
+    app: 'apm',
+    path: 'error_group_overview'
+  });
   useTrackPageview({ app: 'apm', path: 'error_group_overview', delay: 15000 });
 
+  const localUIFiltersConfig = useMemo(
+    () => ({
+      filterNames: [
+        'transactionResult',
+        'host',
+        'containerId',
+        'podId'
+      ] as LocalUIFilterName[],
```

Comment:
I see a bunch of broad type casts. I'm not saying we should never do this but I think we should be cautious when doing it. Eg the following is accepted byt typescript:

```js
      filterNames: [
        'transactionResult',
        'host',
        'containerId',
        'podId1337',
      ] as LocalUIFilterName[],
```
`podId1337` is not a valid filter and should be caught by the compiler.

I suggest this instead:
```
filterNames: ['host' as const, 'containerId' as const, 'podId' as const],
```

Reply:
Thanks, this is super useful. I always struggle with how TS cannot match literal strings to types and this should help. Further reading: https://github.com/Microsoft/TypeScript/issues/26979

Reply:
@sqren this works as well:

```ts
const localUIFiltersConfig: React.ComponentProps<typeof LocalUIFilters> = useMemo(...)
```

The benefit of this is that we don't have to use `as const` everywhere, and we don't need high-fi inferred types here, so the benefits of `as const` are limited. What do you think?

Reply:
I like that! It seems to check that the `filterNames` have the right const values and that all required props are given. One minor glitch is that it doesn't complain about invalid properties:

```ts
  const localUIFiltersConfig: React.ComponentProps<
    typeof LocalUIFilters
  > = useMemo(
    () => ({
      filterNames: ['transactionResult', 'host', 'containerId', 'podId'],
      lolz: true, // this gave an error before
      params: {
        serviceName,
        transactionType
      },
      projection: PROJECTION.ERROR_GROUPS
    }),
    [serviceName, transactionType]
  );
```

Not a big deal though - TS is all about compromises :D

Reply:
@sqren: huh, yeah, seems like a generic factory function allows unknown properties: https://www.typescriptlang.org/play/#code/C4TwDgpgBAKuEEEoF4oG8oDMD22BcUAzsAE4CWAdgOZQC+A3AFCOYCuFAxsGdhVCRGCsSFALIQAPDAB8UABRYAhl2wkQAMQoEFAShSyYUHXkNpGUC-0HC+mZcFUaKcnU1rMOvYlEUn4SVAEhEXE5XX15DBx8KAByWIAaKAAjRQAvAni6HVdGTwpvZL9IAPQsXEzElPTKunogA

I'll just type the return value inside of useMemo.